### ISSUE TITLE
Do release in two stages

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector-releases#773]
To avoid having to re-tag and catch issues earlier, we can have a two stage approach to the release: we first create the artifacts and have them in some sort of 'staging' state, and the actual release step consists in moving said artifacts to the places where users expect them. We could do this with two sets of tags or with a workflow for staging + a tag for the actual release.

One very small step in that direction is #772

**Comment by mx-psi** [comment]
From @douglascamata  on Slack, discussed on the 2025-01-22 SIG meeting:

> here's a link from the goreleaser docs that shows that they have some support for it in the release command: https://goreleaser.com/cmd/goreleaser_release/
--prepare                      Will run the release in such way that it can be published and announced later with goreleaser publish and goreleaser announce (implies --skip=publish,announce,after) (Pro only)

**Comment by mx-psi** [comment]
I think not covering MSI at the start is fine, it is still a huge improvement to do things in two stages for the rest of artifacts

**Comment by douglascamata** [comment]
I agree with @mx-psi. We can start without covering MSI and discuss with the Goreleaser folks if they could support nightlies for MSIs in the future.

Awesome work btw, @mowies. 👏
